Astro 12/12 release notes (#3179)
astronomer/docs: astro/release-notes.md

<!-- ALL LINKS TO INTERNAL DOCS MUST BE COMPLETE URLS INCLUDING HTTPS. Otherwise the links will break in RSS. -->

## December 12, 2023

### Bug fixes

> - Fixed an issue where the Cloud UI would produce an error if you updated an environment variable on an Astro Hybrid Deployment running the Kubernetes Executor

Author: ```suggestion
- Fixed an issue where the Cloud UI would produce an error if you updated an environment variable on an Astro Hybrid Deployment running the Kubernetes Executor.
```

<!-- ALL LINKS TO INTERNAL DOCS MUST BE COMPLETE URLS INCLUDING HTTPS. Otherwise the links will break in RSS. -->

## December 12, 2023

### Additional improvements

> - The Cloud UI Environment Manager now shows a status for whether a connection was successfully synced to Deployments.

Author: ```suggestion
- The Cloud UI Environment Manager now shows a status for whether a connection has been successfully synced to Deployments.
```

> ### Additional improvements
> 
> - The Cloud UI Environment Manager now shows a status for whether a connection has been successfully synced to Deployments.
> 
> ### Bug fixes

Author: ```suggestion
### Bug fixes
```
